march_hardware/src/IMotionCube.cpp

```diff
@@ -467,6 +467,12 @@ bool IMotionCube::goToTargetState(IMotionCubeTargetState targetState)
     this->setControlWord(targetState.getControlWord());
     ROS_INFO_THROTTLE(0.5, "\tWaiting for '%s': %s", targetState.getDescription().c_str(),
                       std::bitset<16>(this->getStatusWord()).to_string().c_str());
+    if (targetState.getDescription() == "Operation Enabled" && this->getState(this->getStatusWord()) == IMCState::fault)
```

Comment:
`targetState.getDescription() == "Operation Enabled"`
should be performed with an enum although I don't understand why it is needed at all?

Reply:
I tried comparing with the enum but wasn't allowed to by the compiler for some reason. Will check later why exactly it wasn't allowed.
Reason why it is necessary at all is that the very first state transition in the goToOperationEnabled is from fault -> switch on disabled. Without the targetState = Operation Enabled check, an error would be thrown while it shouldn't be.

Reply:
What I tried: targetState == IMotionCubeTargetState::OPERATION_ENABLED
What compiler tells me: "error: no match for ‘operator==’ (operand types are ‘march4cpp::IMotionCubeTargetState’ and ‘const march4cpp::IMotionCubeTargetState’)"
Tried googling but couldn't really find a good fix, so then went with the next best thing which was comparing with the string.